### DISCUSSION TITLE
Is it possible at this point to use data provider results as a parameter for another data provider?

### DISCUSSION THREAD

**Discussion by bloomt** [Budibase/budibase#2288] [Help] [answered]
Scenario: My budibase app is connected to a MS SQL server so at this point I am limited to using queries to access the data.

Lets say that I have 3 tables 
Users (Contains UID, Email, etc)
Companies (Contains, CID, ETC)
CompanyAssociation (used as a one to many reference, contains UID and CID)

In my app I have a data provider with a parameter that pulls the email from the current user in budibase and then match it to the email in the Users table. I then have a repeater field connected to the Users data collector which allows me to access the fields within the query result. At this point I am able to get the UID as a single value which I can use as a parameter for further data providers.

I then take the users repeater (which will only return one result) and add another data provider connected to a query of Companies that is expecting the UID as a parameter. This data provider can return 1 or more results. The ultimate goal is to use the CID's that are returned by data provider and use them as the parameter in a query that queries companies and display in a table. As of right now I cannot find a way to do this, I can add another repeater and do the same thing that I did to get the user data but because there can be more than one result I end up with multiple tables each one showing 1 result.

So what I really need to be able to do is access the Data Provider data and use it without using a repeater field.

At this point I'm just not sure that that is possible currently in budibase but I figured that I would ask.

Some other posts hint that pluck handlebar function may be able to do this but I don't know if I can "Pluck" data directly from a data provider.

Any input would be nice, even if its just to tell me that it's not possible at this point.

**Comment by mjashanks** [comment, marked as answer]
Hey @bloomt 

So, please excuse me if I am misunderstanding, but it seems to me like you should be able to do this using:
1. Your users query + repeater returning 1 result (as above)
2. A single SQL query, using your  many-to-many join:

```
SELECT
     Companies.*
FROM CompanyAssociation 
INNER JOIN Companies ON CompanyAssociation.CID = Companies.CID
WHERE CompanyAssociation.UID = {{ UID from Repeater }}
```

3. A Data Provider against this query + a table inside

Is there a reason why you cannot do it this way?


in fact, you may be able to do it without step 1...

```
SELECT
     Companies.*
FROM CompanyAssociation 
INNER JOIN Companies ON CompanyAssociation.CID = Companies.CID
INNER JOIN Users ON CompanyAssociation.UID = Users.UID
WHERE Users.Email = {{ users email }}
```

**Reply by bloomt** [reply]
That works great, guess its time to brush up on my SQL knowledge. Thanks for the quick reply.

**Reply by bloomt** [reply]
Another question that is unrelated. Is it possible to have button on a table that when clicked updates the data provider for another table.

For example lets say I add another table called departments. There are 2 tables on one screen Table one shows companies with a button to "View Departments" when clicked it would take the CID from the row on the first table and use it in a query/data provider for the second table?

**Reply by bloomt** [reply]
Seems like I would need an action that runs a query then drops the result into a variable that I would use in the second data provider. As far as I have seen so far there isn't any way to do this.

**Reply by mjashanks** [reply]
I know what you mean, but not at the moment. For this, I think we would need 2 new action types, that you could add to a button
1. Set a value in the current state = e.g. `Set CurrentCompany to this row._id`
2. Refresh datasource - e.g. `Reload the Departments datasource using CurrentCompany`
